deploy.cake

```diff
@@ -174,13 +175,59 @@ Task("Publish-VstsTask")
     }
 });
 
+
+Task("Publish-DockerImage")
+    .IsDependentOn("DownloadGitHubReleaseArtifacts")
+    .Does(() =>
+{
+    var returnCode = StartProcess("docker", new ProcessSettings
+    {
+        Arguments = "build . --build-arg GitVersionZip=" + artifactLookup["zip"] + " --tag gittools/gitversion"
+    });
+    if (returnCode != 0) {
+        Information("Publish-DockerImage Task failed to build image, but continuing with next Task...");
+        publishingError = true;
+    }
+    
+    // Login to dockerhub
+    returnCode = StartProcess("docker", new ProcessSettings
+    {
+        Arguments = "login -u=\"" + EnvironmentVariable("DOCKER_USERNAME") +"\" -p=\"" + EnvironmentVariable("DOCKER_PASSWORD") +"\""
+    });
+    if (returnCode != 0) {
+        Information("Publish-DockerImage Task failed to login, but continuing with next Task...");
+        publishingError = true;
+    }
+
+    // Publish Tag
+    returnCode = StartProcess("docker", new ProcessSettings
+    {
+        Arguments = "push gittools/gitversion:" + tag
+    });
+    if (returnCode != 0) {
+        Information("Publish-DockerImage Task failed push version tag, but continuing with next Task...");
+        publishingError = true;
+    }
+
+    // Publish latest
+    returnCode = StartProcess("docker", new ProcessSettings
```

Comment:
Any reason for not using the Cake.Docker addin for this?

https://github.com/MihaMarkic/Cake.Docker